examples/demo-nft-module/README.md

```diff
@@ -0,0 +1,481 @@
+# How to create new module for Sovereign Rollup
+
+
+# Getting Started
+
+Module is a separate crate that needs to implement `sov_modules_api::Module` trait and it's own change logic based on input messages.
+
+Let's start with defining basic dependencies in `Cargo.toml`:
+
+```toml
+[dependencies]
+anyhow = { anyhow = "1.0.62" }
+sov-modules-api = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main", default-features = false }
+sov-modules-macros = { git = "https://github.com/Sovereign-Labs/sovereign.git", branch = "main" }
+```
+
+## Module definition
+
+NFT module is defined as following:
+
+```rust
+use sov_modules_api::Context;
+use sov_modules_macros::ModuleInfo;
+
+#[derive(ModuleInfo, Clone)]
+pub struct NonFungibleToken<C: Context> {
+    #[address]
+    pub address: C::Address,
+
+    #[state]
+    pub(crate) admin: sov_state::StateValue<C::Address>,
+
+    #[state]
+    pub(crate) owners: sov_state::StateMap<u64, C::Address>,
+}
+```
+
+Where it has:
+
+1. Address. Each module has to have an address. A module address is similar to a smart contract address in Ethereum. For example, a module can receive tokens. The address is calculated from a hash of the module prefix. This means that:
+    - The module address is unique
+    - No one knows the private key that produces this address
+
+2. Several state attributes. In this case it is an address of admin, and map of token id to owner address. Token id is u64 for simplicity.
+
```

Comment:
It would be good to mention modules can also use `#[module]` to store a reference to another module, even though we don't use that feature in this example.

Reply:
Done